### PR TITLE
feat(nuxt): define a custom root element inside `app.config`

### LINKED ISSUE #8603: Customize app root element
### Describe the feature

Currently, the Nuxt app is mounted inside a `#__nuxt` container and this cannot be changed. Allowing for a custom root element would allow for:
- customizing the rendered HTML page using a Nitro plugin and mount the app inside that custom page;
- creating a no-SSR app and incorporating it into an existing HTML page, but with all the benefits that Nuxt brings along.

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://v3.nuxtjs.org/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/framework/discussions) and [issues](https://github.com/nuxt/framework/issues).

### PR DESCRIPTION
**Pull request by yassilah**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Resolves #8603 and #8099.
There's probably a better place to customize the root element selector than `app.config` but it allowed for a very simple implementation and for retrieving that root container from wherever we want inside the app using `useAppConfig()`, which might be useful in some cases 

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by HigherOrderLogic** [comment]
It would be nice if you also add an option to change the `window.__NUXT__` script to something like `window.__FOO__`.

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-lockfileVersion: 5.4
+lockfileVersion: 5.3
```
Comment (Aareksio) [inline pnpm-lock.yaml:1]: Shouldn't be a part of this PR

**Comment by manniL** [comment]
> It would be nice if you also add an option to change the `window.__NUXT__` script to something like `window.__FOO__`.

I don't think this will be feasible to maintain, see https://github.com/nuxt/framework/issues/8099

**Review by mohsin — APPROVED** [review]
lgtm

**Comment by pi0** [comment]
Thanks for PR @yassilah ❤️ I've made a slightly different implementation that we are also able to change renderer behavior based on configuration: #8883

**Comment by yassilah** [comment]
that's much better, this way i don't even need a nitro plugin to render my custom div!

**Pull request closed without merging**